src/domain/index.ts

```diff
@@ -208,3 +208,21 @@ export type Gas =
         gasPrice: BigNumber;
       };
     };
+
+export type TokenSpendPermission =
+  | {
+      type: "allowed";
```

Comment:
I would replace this `allowed` discriminator with `none`, which I think it's more aligned with the name of the type.

Reply:
This was initially `non-required` 🤣

Reply:
Mmmm I don't see it that way, because at the end when you see this type, you ask yourself: "Which permission do I need to be able to spend the token?". You probably need an `approval` permission, or maybe a `permit` permission. Or maybe you don't need a permission at all, which should be `none`. But if you ask this yourself and you see `allowed` looks quite weird.

In fact, I think that all the keys of this `Permission` should be nouns: `none`, `approval`, `permit`.

Reply:
> In fact, I think that all the keys of this `Permission` should be nouns: `none`, `approval`, `permit`.

There you got me. Done.